src/libstd/sync/once.rs

```diff
@@ -35,23 +85,44 @@ use sync::StaticMutex;
 /// ```
 #[stable(feature = "rust1", since = "1.0.0")]
 pub struct Once {
-    mutex: StaticMutex,
-    cnt: AtomicIsize,
-    lock_cnt: AtomicIsize,
+    state: AtomicUsize,
```

Comment:
This struct should have some `PhantomData` describing the kind of data `state` can point to. (Basically this should be done any time a pointer is hidden in a usize, so that auto traits etc work correctly.)